### ISSUE TITLE
access playing time - feature request

### ISSUE THREAD

**Issue by pennyscan** [mpv-player/mpv#4653]
Any platform or version

A simple way to access time and duration...
With Media player classic or pot player we can either:

1) assign a key so player copies current time / duration to clipboard

   press key x                ( clipboard is now   01:25 / 11:16 )

2) or use windows messages, (better) eg with autohotkey

   ControlGetText, time, Static2, ahk_class mpv  
   (time variable is now   01:25 / 11:16)

It may be possible using lua and socat and json ipc, but these technologies may be too complicated for a lot of potential developers

**Comment by wiiaboo** [comment]
Press `o` or `shift+p`?

**Comment by Argon-** [comment]
On OSX you can create a keybinding like this:

    l run "/bin/sh" "-c" "echo ${playback-time} | pbcopy"

This should be possible for other systems as well when you swap the pasteboard program with whatever your system is using.

**Comment by wiiaboo** [comment]
Here's some [lua code](https://github.com/wiiaboo/mpv-scripts/blob/master/excerpt.lua#L69-L80) to copy some text to clipboard. Right below there's also code that does what you probably want.

You would place this in a script, add a couple of lines to get current time/duration and then a key binding to copy those to clipboard.

This, of course, only works for Windows. For different systems there's different ways to use the clipboard.

**Comment by pennyscan** [comment]
thanks very much for your help, 
I'll try testing the sample code you gave tomorrow.

**Comment by garoto** [comment]
For Windows Vista and later that ships with `clip.exe`, this works:
```
<KEY> run "cmd.exe" "/d" "/c" "echo ${playback-time} / ${duration}|clip"
```

**Comment by medhefgo** [comment]
Maybe mpv should provide a set-clipboard command, then anyone can easily rebind their command to use it:
`o show-text ${playback-time}; set-clipboard ${playback-time}`

Would have the benefit of being cross platform and not requiring to run extra binaries just to clipboard something.

**Comment by Argon-** [comment]
This command would be cross platform because then mpv would have to maintain platform-specific implementations for it.
Personally, I don't see why a movie player would implement platform-specific clipboard code, especially when it's so easy to make a "run shell" implementation.

**Comment by pennyscan** [comment]
you might be onto something. This works pefectly with ffmpeg...

       Run %COMSPEC% /c ffmpeg.exe -i test.mp4 2>&1 | find "Duration" | Clip, , hide && exit

so are you saying mpv can do something like this too ?
I have tried a few options, but can't get it to work.

**Comment by pennyscan** [comment]
thanks wiiaboo, but i could not get the lua code to work
this is my input file..

local function set_clipboard(text)
    local res = utils.subprocess({ args = {
        'powershell', '-NoProfile', '-Command', string.format([[& {
            Trap {
                Write-Error -ErrorRecord $_
                Exit 1
            }
            Add-Type -AssemblyName PresentationCore
            [System.Windows.Clipboard]::SetText('%s')
        }]], text)
    } })
end

mp.add_key_binding("5", "set_clipboard",
function () set_clipboard("hello world") end)


thanks medhefgo

i tried
p::  
    run "cmd.exe" "/d" "/c" "echo ${playback-time} / ${duration}|clip"
and
   Run %COMSPEC% /c "cmd.exe" "/d" "/c" "echo ${playback-time} / ${duration}|clip"

neither seem to work
but
Run %COMSPEC% /c ffmpeg.exe -i test.mp4 2>&1 | find "Duration" | Clip, , hide && exit
works ok for ffmpeg

**Comment by garoto** [comment]
You're supposed to add the `run` line to your `input.conf` file located at `%appdata\mpv\`. Create one (input.conf) if it doesn't exist already.

It should read something like this after:

![image](https://user-images.githubusercontent.com/11768175/28572642-43dba59a-711e-11e7-802b-f531cb466841.png)

Lua scripts should go into a `scripts` dir at the same %appdata% location and have a .lua extension.

**Comment by pennyscan** [comment]
Thanks garoto, that totally worked now.

I really appreciate all your help guys.

**Comment by TunnelThruTime** [comment]
I've tried everything that's been written here with absolutely no success! it's a little ridiculous!! 
I'm not a developer and am rather a user, but I feel it should be allot easier to write a single line of code for this kind of thing. 
That is after all what everyone on the thread is saying. 
Instead for the:
`l run "/bin/sh" "-c" "echo ${playback-time} | pbcopy"`
 `"subprocess failed: init"`
I've also attempted: 
`echo 'show-text ${playback-time}' | socat - /tmp/mpvsocket`

**Comment by wm4** [comment]
Are you on OSX? The first method works for me with xclip on x11. The second method also works (of course that requires enabling IPC).

**Comment by Akemi** [comment]
both methods also work for me on macOS.

**Comment by bharathyes** [comment]
> You're supposed to add the `run` line to your `input.conf` file located at `%appdata\mpv\`. Create one (input.conf) if it doesn't exist already.

> ![image](https://user-images.githubusercontent.com/11768175/28572642-43dba59a-711e-11e7-802b-f531cb466841.png)


Copying to clipboard works fine on Windows10, mpv installed via chocolatey.

But I want the timestamp to be appended to a particular text file, so I can use it at a later point instead of having to switch apps and paste.

I tried this: `t run "cmd.exe" "/d" "/c" "echo ${playback-time}>>D:\mpv\trim-video.txt"`

Is there any obvious errors with this command? It worked for a couple of days. I used it more than a dozen times. But now it doesn't work and I can't figure out what changed.
